hooks/pre-commit

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh -x
+#
+# Recommended pre-commit git hook for activity-stream github repo
+#
+# Install by executing
+#
+#   ln -s ../../hooks/pre-commit .git/hooks/pre-commit
+#
+# at the top-level of the activity-stream github repo.
+#
+# Runs `eslint --fix` on all selected files, which, given our current
+# prettier configuration, means prettifying these files as well.  The
+# commit will be aborted if eslint exits with a failure code.
+#
+# Based on the example script in the prettier docs at
+# https://prettier.io/docs/en/precommit.html
+
+FILES=$(git diff --cached --name-only --diff-filter=ACM "*.js" "*.jsx" "*.jsm" | sed 's| |\\ |g')
+echo "files to check: $FILES"
+[ -z "$FILES" ] && exit 0
+
+echo "$FILES" | xargs -t ./node_modules/.bin/eslint --fix
```

Comment:
Does --cache here help? Theoretically people might have already run test/lint/fix before trying to commit which should have updated .eslintcache ?

Reply:
Shouldn't hurt; I'll add it.

Reply:
Done